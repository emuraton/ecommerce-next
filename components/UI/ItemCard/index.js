import React from 'react';

import Card from './styles';

const ItemCard = ({ onClick }) => {
  return (
    <Card.Container>
      <Card.ImageContainer>
        <Card.Image src="/static/deluxe-room.jpg" />
      </Card.ImageContainer>
      <Card.FlexColumn>
        <Card.H2>Deluxe Room</Card.H2>
        <Card.List>
          <Card.ListItem>39 squares meters</Card.ListItem>
          <Card.ListItem>Free standart WiFi access</Card.ListItem>
          <Card.ListItem>Free entry to the Infinity Pool</Card.ListItem>
        </Card.List>
        <div>
          <Card.Price>$108</Card.Price>
          <Card.PerNight>per night</Card.PerNight>
        </div>
        <div>
          <Card.Price disabled>$158</Card.Price>
          <Card.PerNight disabled>per night</Card.PerNight>
        </div>
        <Card.Button type="button" onClick={onClick}>
          Book Now
        </Card.Button>
      </Card.FlexColumn>
    </Card.Container>
  );
};

export default ItemCard;
