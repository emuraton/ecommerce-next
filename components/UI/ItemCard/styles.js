import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

const Container = styled.section`
  display: flex;
`;

const ImageContainer = styled.div`
  width: 50%;
  margin-right: 20px;
`;

const Image = styled.img`
  height: 250px;
  width: 100%;
  object-fit: cover;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
`;

const List = styled.ul`
  padding-left: 10px;
  margin-top: 10px;
`;

const ListItem = styled.li`
  font-weight: 500;
  color: rgb(72, 72, 72);
  margin-top: 2px;
  margin-left: 4px;
  margin-bottom: 0px;
  font-size: 15px;
  line-height: 18px;
`;

const H2 = styled.h2`
  font-weight: 500;
  margin: 5px 0;
`;

const FlexColumn = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
`;

const Price = styled.span`
  font-size: 22px;
  font-weight: 600;
  ${ifProp(
    'disabled',
    css`
      font-size: 20px;
      font-weight: 600;
      opacity: 0.5;
      text-decoration: line-through;
    `,
  )};
`;

const PerNight = styled.span`
  margin-left: 3px;
  font-size: 12px;

  ${ifProp(
    'disabled',
    css`
      opacity: 0.5;
      text-decoration: line-through;
    `,
  )};
`;

const SubmitButton = styled.button`
  position: absolute;
  bottom: 9px;
  left: -5px;
  width: 100%;
  height: 40px;
  margin-top: 20px;
  background-color: rgb(0, 132, 137);
  color: white;
  border-radius: 3px;
  outline: none;
  border-color: transparent;
  font-size: 18px;
`;
SubmitButton.displayName = 'Button';

const Card = {
  Button: SubmitButton,
  Price,
  PerNight,
  FlexColumn,
  H2,
  List,
  ListItem,
  Container,
  Image,
  ImageContainer,
};

export default Card;
