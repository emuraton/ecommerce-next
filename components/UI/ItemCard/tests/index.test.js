import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ItemCard from '../';

describe('<ItemCard />', () => {
  const initialProps = {
    onClick: Function.prototype,
  };
  describe('@renders', () => {
    it('default render', () => {
      const wrapper = shallow(<ItemCard {...initialProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('@events', () => {
    // TODO refactor test once onClick actually do something
    it('onClick', () => {
      const mockedOnClick = jest.fn();
      const props = {
        ...initialProps,
        onClick: mockedOnClick,
      };
      const wrapper = shallow(<ItemCard {...props} />);
      const button = wrapper.find('Button');
      expect(mockedOnClick).toHaveBeenCalledTimes(0);
      button.simulate('click');
      expect(mockedOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
