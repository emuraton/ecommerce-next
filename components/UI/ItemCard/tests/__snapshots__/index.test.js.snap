// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemCard /> @renders default render 1`] = `
<styles__Container>
  <styles__ImageContainer>
    <styles__Image
      src="/static/deluxe-room.jpg"
    />
  </styles__ImageContainer>
  <styles__FlexColumn>
    <styles__H2>
      Deluxe Room
    </styles__H2>
    <styles__List>
      <styles__ListItem>
        39 squares meters
      </styles__ListItem>
      <styles__ListItem>
        Free standart WiFi access
      </styles__ListItem>
      <styles__ListItem>
        Free entry to the Infinity Pool
      </styles__ListItem>
    </styles__List>
    <div>
      <styles__Price>
        $108
      </styles__Price>
      <styles__PerNight>
        per night
      </styles__PerNight>
    </div>
    <div>
      <styles__Price
        disabled={true}
      >
        $158
      </styles__Price>
      <styles__PerNight
        disabled={true}
      >
        per night
      </styles__PerNight>
    </div>
    <Button
      onClick={[Function]}
      type="button"
    >
      Book Now
    </Button>
  </styles__FlexColumn>
</styles__Container>
`;
